contrib/initramfs/hooks/zfsunlock

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+
+PREREQ="dropbear"
+
+prereqs() {
+    echo "$PREREQ"
+}
+
+case "$1" in
+    prereqs)
+        prereqs
+        exit 0
+    ;;
+esac
+
+. /usr/share/initramfs-tools/hook-functions
+
+if [ -r "${DESTDIR}/sbin/dropbear" ]; then
+	cat >"$DESTDIR"/bin/zfsunlock <<'EOF'
+#!/bin/sh
+while [ ! -e /run/zfs_fs_name ] && [ ! -e /run/zfs_unlock_complete ]; do
+	if [ ! $entry_msg_shown ]; then
```

Comment:
Why is it necessary for the initial message to be inside the loop? Given the conditions of this loop and what is going on with those files, it's always going to fire. You could eliminate `entry_msg_show entirely` and just put the echo above the loop.